### PR TITLE
Fix interpreter cannot call "imported-and-exported-back" function

### PR DESCRIPTION
**Pull request by nullpo-head**
This PR fixes a bug that interpreter crashes when it calls such a guest function that was imported from the host into a wasm module, but was exported to the host again. This is an interpreter-only bug and JIT has been correctly handling this without problem.

Please note that this PR alone will not allow us to test the WASI APIs by calling such "imported and exported back" WASI API functions. It turned out that there is another issue. See #195

### CODE REVIEW

File: tests/engine/adhoc_test.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func runTests(t *testing.T, newEngine func() wasm.Engine) {
 	unreachable(t, newEngine)
 	memory(t, newEngine)
 	recursiveEntry(t, newEngine)
+	importedAndExportedFunc(t, newEngine)
```
Comment (nullpo-head) [inline tests/engine/adhoc_test.go:35]: Can we rewrite here like the following? Do we intentionally avoid it or is it ok to do that?
```go

	t.Run("fibonacci", func(t *testing.T) {
		fibonacci(t, newEngine)
	})
	t.Run("fac", func(t *testing.T) {
		fac(t, newEngine)
	})
	t.Run("unreachable", func(t *testing.T) {
		unreachable(t, newEngine)
	})
	t.Run("memory", func(t *testing.T) {
		memory(t, newEngine)
	})
	t.Run("recursive entry", func(t *testing.T) {
		recursiveEntry(t, newEngine)
	})
	t.Run("imported-and-exported func", func(t *testing.T) {
		importedAndExportedFunc(t, newEngine)
	})
```

Comment (codefromthecrypt) [reply tests/engine/adhoc_test.go:36]: sounds good

File: tests/engine/adhoc_test.go [hunk 2]
```diff
@@ -170,3 +172,28 @@ func recursiveEntry(t *testing.T, newEngine func() wasm.Engine) {
 	_, _, err = store.CallFunction("test", "main", uint64(1))
 	require.NoError(t, err)
 }
+
+func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
+	buf, err := os.ReadFile("testdata/imported_and_exported_func.wat")
```
Comment (codefromthecrypt) [inline tests/engine/adhoc_test.go:177]: maybe just backtrack this as for small text inlining > go:embed > os.ReadFile

File: tests/engine/adhoc_test.go [hunk 3]
```diff
@@ -170,3 +172,28 @@ func recursiveEntry(t *testing.T, newEngine func() wasm.Engine) {
...
+	require.NoError(t, err)
+	mod, err := text.DecodeModule(buf)
+	require.NoError(t, err)
+
+	store := wasm.NewStore(newEngine())
+
+	// addInt is imported by the wasm module, and the wasm module exports addInt back
```
Comment (codefromthecrypt) [inline tests/engine/adhoc_test.go:184]: ex if you inline you will already have the comments about what's going on in the source in the same test

**Review by codefromthecrypt — APPROVED** [review]
some style suggestions, thanks for fixing the first bug in the import-and-export-back!

**Review by mathetake — APPROVED** [review]
Awesome!

File: wasm/interpreter/interpreter.go [hunk 4]
```diff
@@ -477,12 +477,12 @@ func (it *interpreter) Call(f *wasm.FunctionInstance, params ...uint64) (results
 		return
 	}
 
+	for _, param := range params {
+		it.push(param)
+	}
 	if g.hostFn != nil {
 		it.callHostFunc(g, params...)
```
Comment (mathetake) [inline wasm/interpreter/interpreter.go:484]: Could you delete `_ ...uint64` from `callHostFunc`'s signature? https://github.com/tetratelabs/wazero/blob/07248537ca015313b4dcc16e2507f4905c8eac91/wasm/interpreter/interpreter.go#L495

Comment (nullpo-head) [reply wasm/interpreter/interpreter.go:484]: sur

File: tests/engine/adhoc_test.go (hunk 2 above)
Comment (nullpo-head) [reply tests/engine/adhoc_test.go]: inlined!

**Pull request merged**